File: GullSharksWeb/src/models/Order.ts
import { OrderDetail } from "./OrderDetail";
import { PaymentDetail } from "./PaymentDetail";
import { ShippingAddress } from "./ShippingAddress";

export interface Order {
  id: number;
  game_ID: number;
  orderDetail_ID: number;
  user_ID: number;
  isConfirmed: boolean;
  isDeleted: boolean;
  paymentDetails?: PaymentDetail,
  orderDetails?: OrderDetail,
  shippingAddress?: ShippingAddress
  orderName?: string;
}

File: GullSharksLib/Models/Order.cs
namespace GullSharksLib;

public class Order
{
    public int ID { set; get; }
    public int Game_ID { set; get; }
    public double Subtotal { set; get; }
    public double Total { set; get; }
    public bool IsConfirmed { get; set; }
    public DateTime DateCreated { set; get; }
    public int OrderDetail_ID { set; get; }
    public int User_ID { set; get; }
    public bool IsDeleted { get; set; }

}